resources/js/Wishlist.vue

```diff
@@ -0,0 +1,199 @@
+            });
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            }
+        },
+
+        async removeItem(wishlist, wishlistItemId, redirect) {
+            await this.$root.apiRequest('delete', '/api/wishlists/item/' + wishlistItemId, {}, function (response) {
+                window.Vue.delete(wishlist.items, wishlist.items.findIndex(e => e.wishlist_item_id == wishlistItemId));
+            });
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            }
+        },
+
+        async editItem(wishlist, wishlistItemId, variables, redirect) {
+            await this.$root.apiRequest('patch', '/api/wishlists/item/' + wishlistItemId, variables, function (response) {
+                var item = wishlist.items.find(e => e.wishlist_item_id == wishlistItemId);
+                item.description = response.data.description;
+                item.qty = response.data.qty;
+            })
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            }
+        },
+
+
+        async addWishlist(title, redirect) {
+            var wishlists = this.wishlists;
+            var ret = null;
+            await this.$root.apiRequest('post', '/api/wishlists/', {
+                title: title
+            }, function (response) {
+                response.data.items = [];
+                wishlists.push(response.data);
+                ret = response.data;
+            });
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            } else {
+                return ret;
+            }
+        },
+
+        async editWishlist(wishlist, title, description, shared, redirect) {
+            await this.$root.apiRequest('patch', '/api/wishlists/' + wishlist.id, {
+                title: title,
+                description: description ?? '',
+                shared: shared
+            }, function (response) {
+                wishlist.title = response.data.title;
+                wishlist.description = response.data.description;
+                wishlist.shared = response.data.shared;
+            });
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            }
+        },
+
+        async removeWishlist(wishlist, redirect) {
+            var wishlists = this.wishlists;
+            await this.$root.apiRequest('delete', '/api/wishlists/' + wishlist.id, {}, function (response) {
+                window.Vue.delete(wishlists, wishlists.findIndex(e => e.id == wishlist.id));
+            })
+            this.updateLocalStorage();
+
+            if(redirect) {
+                Turbolinks.visit(redirect);
+            }
+        },
+
+        updateLocalStorage() {
+            localStorage.wishlists = JSON.stringify(this.wishlists);
```

Comment:
Watching the data is maybe a better idea so you don't have to call this method each time. Also have a look at: https://vueuse.org/core/uselocalstorage/, we're planning to use that in the Rapidez Core.